src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -201,21 +201,36 @@ def generate_pex_arg_list(self) -> List[str]:
             args.extend(["--interpreter-constraint", str(constraint)])
         return args
 
+    def _includes_version(self, major_minor: str, last_patch: int) -> bool:
+        patch_versions = list(reversed(range(0, last_patch + 1)))
+        for req in self:
+            if any(
+                req.specifier.contains(f"{major_minor}.{p}") for p in patch_versions  # type: ignore[attr-defined]
+            ):
+                return True
+        return False
+
     def includes_python2(self) -> bool:
         """Checks if any of the constraints include Python 2.
 
         This will return True even if the code works with Python 3 too, so long as at least one of
         the constraints works with Python 2.
         """
-        py27_patch_versions = list(
-            reversed(range(0, 18))
-        )  # The last python 2.7 version was 2.7.18.
-        for req in self:
-            if any(
-                req.specifier.contains(f"2.7.{p}") for p in py27_patch_versions  # type: ignore[attr-defined]
-            ):
-                return True
-        return False
+        return self._includes_version("2.7", last_patch=18)
+
+    def minimum_python_version(self) -> Optional[str]:
+        """Find the lowest major.minor Python version that will work with these constraints.
+
+        The constraints may also be compatible with later versions; this is the lowest version that
+        still works.
+        """
+        if self.includes_python2():
+            return "2.7"
+        for major_minor in ("3.5", "3.6", "3.7", "3.8", "3.9", "3.10"):
+            # Assume any 3.x release has no patch after 12. The max is currently 9.
+            if self._includes_version(major_minor, last_patch=12):
```

Comment:
I might break this out into a named constant `_MAX_SUPPORTED_PATCH_VERSION = 12` at the module or class level?